HeavyFlavorAnalysis/RecoDecay/src/BPHDecayVertex.cc

```diff
@@ -122,34 +126,42 @@ const reco::Vertex& BPHDecayVertex::vertex(VertexFitter<5>* fitter,
 
 const vector<const reco::Track*>& BPHDecayVertex::tracks() const {
   if (oldTracks)
-    tTracks();
+    fTracks();
   return rTracks;
 }
 
 const reco::Track* BPHDecayVertex::getTrack(const reco::Candidate* cand) const {
   if (oldTracks)
-    tTracks();
+    fTracks();
   map<const reco::Candidate*, const reco::Track*>::const_iterator iter = tkMap.find(cand);
   map<const reco::Candidate*, const reco::Track*>::const_iterator iend = tkMap.end();
   return (iter != iend ? iter->second : nullptr);
 }
 
-const vector<reco::TransientTrack>& BPHDecayVertex::transientTracks() const {
+char BPHDecayVertex::getTMode(const reco::Candidate* cand) const {
   if (oldTracks)
-    tTracks();
+    fTracks();
+  map<const reco::Candidate*, char>::const_iterator iter = tmMap.find(cand);
+  map<const reco::Candidate*, char>::const_iterator iend = tmMap.end();
+  return (iter != iend ? iter->second : '.');
+}
+
+const vector<reco::TransientTrack>& BPHDecayVertex::transientTracks() const {
+  if (oldTTracks)
+    fTTracks();
   return trTracks;
 }
 
 reco::TransientTrack* BPHDecayVertex::getTransientTrack(const reco::Candidate* cand) const {
-  if (oldTracks)
-    tTracks();
+  if (oldTTracks)
+    fTTracks();
   map<const reco::Candidate*, reco::TransientTrack*>::const_iterator iter = ttMap.find(cand);
   map<const reco::Candidate*, reco::TransientTrack*>::const_iterator iend = ttMap.end();
   return (iter != iend ? iter->second : nullptr);
 }
 
 /// retrieve EventSetup
-const edm::EventSetup* BPHDecayVertex::getEventSetup() const { return evSetup; }
+const BPHEventSetupWrapper* BPHDecayVertex::getEventSetup() const { return evSetup; }
 
 const string& BPHDecayVertex::getTrackSearchList(const reco::Candidate* cand) const {
   static string dum = "";
```

Comment:
No reason to make it `static` (as also pointed out by the static analyzer)
Why not simply `return "";` at L 171?

Reply:
The functions return by reference, if not making that static or doing a
`return ""`
the compiler complains about "returning reference to temporary"

Reply:
Correct...
What about `static const` ?